clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10619,36 +10608,66 @@ static void getOtherSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
 }
 
 // Add any sycl-post-link options that rely on a specific Triple.
-static void
-getTripleBasedSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
-                               const llvm::opt::ArgList &TCArgs,
-                               llvm::Triple Triple, ArgStringList &PostLinkArgs,
-                               bool SpecConsts, types::ID OutputType) {
+static void getTripleBasedSYCLPostLinkOpts(
+    const ToolChain &TC, const JobAction &JA, const llvm::opt::ArgList &TCArgs,
+    llvm::Triple Triple, ArgStringList &PostLinkArgs, bool SpecConsts,
+    types::ID OutputType, bool NewOffloadDriver = false) {
+  // Note: Do not use Triple when NewOffloadDriver is 'true'.
+  if (!NewOffloadDriver && (OutputType == types::TY_LLVM_BC)) {
+    // single file output requested - this means only perform necessary IR
+    // transformations (like specialization constant intrinsic lowering) and
+    // output LLVMIR
+    addArgs(PostLinkArgs, TCArgs, {"-ir-output-only"});
+  }
+  // specialization constants processing is mandatory
+  if (SpecConsts)
+    addArgs(PostLinkArgs, TCArgs, {"-spec-const=native"});
+  else
+    addArgs(PostLinkArgs, TCArgs, {"-spec-const=emulation"});
 
   // See if device code splitting is requested.  The logic here works along side
   // the behavior in setOtherSYCLPostLinkOpts, where the option is added based
   // on the user setting of-fsycl-device-code-split.
-  if (!(TCArgs.hasArg(options::OPT_fsycl_device_code_split_EQ) ||
-        Triple.getArchName() == "spir64_fpga"))
-    addArgs(PostLinkArgs, TCArgs, {"-split=auto"});
+  if (!NewOffloadDriver) {
+    if (!(TCArgs.hasArg(options::OPT_fsycl_device_code_split_EQ) ||
+          Triple.getArchName() == "spir64_fpga"))
```

Comment:
Not a fan of the `if/else` usage with `NewOffloadDriver` due to the duplication of the option checks.  Is it possible to add some kind of lambda function that will always return `true` for `NewOffloadDriver`, otherwise evaluate the 'triple' check?

Reply:
Thanks for the feedback. I modified the checks here to remove any duplication here. Boolean Algebra came in handy!!